lib/headcount_analyst.rb

```diff
@@ -1,109 +1,82 @@
-require_relative 'district_repository'
-require_relative 'enrollment'
-require_relative 'enrollment_repository'
-require 'pry'
-
 class HeadcountAnalyst
-  attr_reader :dr
+  attr_reader :district_repository
 
-  def initialize(dr)
-    @dr = dr
+  def initialize(district_repository)
```

Comment:
Many of the methods in this class were duplicating code, addressing hashed values and creating a largely unmaintainable and unclear class. Mathematical functions and single responsibility aside, these methods were condensed and parameters are being passed in for clarity.